File: c/p29/p29.c
#include <stdio.h>
#define MAXLIM 100
int ipow(int, int);
int main(void)
{
  int i, j, a, count=0;
  for (i=2;i<=MAXLIM;i++)
    {
      for (j=2;j<=MAXLIM;j++)
	{
	  count++;
	  for (a=2; (a*j)<=MAXLIM; a++)
	    count--;
	}
    }
  printf("%d\n", count);


}
int ipow(int a, int b)
{
  if(a<=0) return 0;
  else if (b <= 0) return 1;
  else return a * ipow(a, b-1);
}

File: Fortran/p29/p29.f90
program p29
  integer, parameter :: MAXLIM=100;
  logical sieve(2:MAXLIM, 2:MAXLIM)
  !sieve = .TRUE.
  icounter = 0
  do 1 i=2,MAXLIM
     do 1 j=2,MAXLIM
        icounter = icounter + 1
        if (i*i > MAXLIM) cycle
        do 1 k=2,j-1
           if (i**k>MAXLIM)exit
           if(mod(j,k)/=0)cycle
           m=j/k ! i^j = i^(m*k) = (i^k)^m

           write(*,10) i, j, k, i**k, m
10         format(i0,'^',i0, ' == (',i0,')',i0,'^',i0)
           !sieve(i**k,m)=.FALSE.
           icounter = icounter - 1;
1         continue
           
   print*, icounter !count(sieve)
           
           
end program
